js/url.ts

```diff
@@ -234,17 +254,24 @@ export class URL {
 
     if (urlParts.protocol) {
       this._parts = urlParts;
+      this.protocol = urlParts.protocol;
+      this.username = urlParts.username;
+      this.password = urlParts.password;
+      this.hostname = urlParts.hostname;
+      this.port = urlParts.port;
+      this.pathname = urlParts.pathname;
+      this.query = urlParts.query;
+      this.hash = urlParts.hash;
     } else if (baseParts) {
-      this._parts = {
-        protocol: baseParts.protocol,
-        username: baseParts.username,
-        password: baseParts.password,
-        hostname: baseParts.hostname,
-        port: baseParts.port,
-        path: urlParts.path || baseParts.path,
-        query: urlParts.query || baseParts.query,
-        hash: urlParts.hash
-      };
+      this._parts = initializedURLParts;
```

Comment:
Why is this necessary?  How is this related to the issue mentioned in the PR?

Reply:
Also, this means every instance of `URL` that does not have protocol, but has `baseParts` would share the same instance of `this._parts` which sounds very bad.

Reply:
in the beginning, I thought the same but turned out setter will refuse provided value if there is no initialized Object in `_parts`.